shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,58 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.repository.exception.ProductNotFoundException;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.repository.ProductRepository;
+
+import java.util.List;
+
+@Service
+public class ProductService {
+    private final ProductRepository productRepository;
+    private final CategoryService categoryService;
+    private final SupplierService supplierService;
+
+    public ProductService(ProductRepository productRepository, CategoryService categoryService, SupplierService supplierService) {
```

Comment:
You can use @RequiredArgsConstructor from Lombok as well. Also in the other services